### PR TITLE
Delay long list arguments

### LINKED ISSUE #4734: Extract long lists passed as arguments into separate tasks
In #4727 it was identified that a long list included in many tasks led to slowdowns for 3 reasons:

- Longer optimization time (need to traverse the list looking for tasks multiple times)
- Longer execution time (scheduler traverses lists in tasks during execution)
- Longer serialization time (including large data in each task is a general no-no).

We already pull out arguments larger than 1e6 bytes into a separate task (wrapped in delayed), but for lists which are interpreted by the optimization/scheduling passes this threshold should likely be lower. If it's not already done, this should also be done for the other collections.

### PR DESCRIPTION
**Pull request by mrocklin**
This turns long lists into separate tasks to avoid repeated
serialization and traversal times.

Fixes #4734

- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @jcrist

File: dask/array/core.py [hunk 1]
```diff
@@ -299,6 +299,8 @@ def normalize_arg(x):
         return x
     elif isinstance(x, str) and re.match(r'_\d+', x):
         return delayed(x)
+    elif isinstance(x, list) and len(x) >= 10:
+        return delayed(x)
```
Comment (jcrist) [inline dask/array/core.py]: Should we add this to `map_partitions` as well?

Comment (mrocklin) [reply dask/array/core.py]: Done

**Comment by jcrist** [comment]
Looks good to me, merging. Thanks @mrocklin.

**Pull request merged**